### PR TITLE
Add promise rejection handler for custom brand image retrieval

### PR DESCRIPTION
**Pull request by jayaddison-collabora**
#### Summary
Adds a promise rejection handler for retrieval of custom brand images.

Errors are handled by setting the presence of a custom brand image (`brandImageExists`) to `false`.  This means that an option to remove the brand image will not be presented to the user.

#### Ticket Link
Resolves mattermost/mattermost-server#17911.

#### Related Pull Requests
N/A

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/admin_console/brand_image_setting/brand_image_setting.jsx [hunk 1]
```diff
@@ -69,7 +69,7 @@ export default class BrandImageSetting extends React.PureComponent {
                     this.setState({brandImageExists: false});
                 }
             },
-        );
+        ).catch(() => this.setState({brandImageExists: false}));
```
Comment (jayaddison-collabora) [inline components/admin_console/brand_image_setting/brand_image_setting.jsx:72]: Could: add a `console.log` statement here as well?

**Comment by Willyfrog** [comment]
hey there! thanks for posting and issue and a fix to solve it, that's amazing  :) . 

Looks like there is a conflict with master can you fix it?

File: components/admin_console/brand_image_setting/brand_image_setting.tsx [hunk 2]
```diff
@@ -78,6 +78,9 @@ export default class BrandImageSetting extends React.PureComponent<Props, State>
             } else {
                 this.setState({brandImageExists: false});
             }
+        }).catch(() => {
+            console.log('unable to retrieve brand image'); //eslint-disable-line no-console
```
Comment (hahmadia) [inline components/admin_console/brand_image_setting/brand_image_setting.tsx:82]: Can we make the following small modification?
```suggestion
        }).catch((err) => {
            console.error(`unable to retrieve brand image: ${err}`); //eslint-disable-line no-console
```

**Review by hahmadia — CHANGES_REQUESTED** [review]
Good job! Minor suggestion

File: components/admin_console/brand_image_setting/brand_image_setting.tsx (hunk 2 above)
Comment (jayaddison-collabora) [reply components/admin_console/brand_image_setting/brand_image_setting.tsx]: Very good suggestion, thank you @hahmadia.  Without that, this change could have inadvertently suppressed fetch error messages on the console, I think; that wasn't my intent!

**Review by Willyfrog — APPROVED** [review]
LGTM! thanks!

**Review by hahmadia — APPROVED** [review]
LGTM THANKS!

**Review by jgilliam17 — APPROVED** [review]
Thanks @jayaddison-collabora 
Tested, looks good to merge.
- Smoke tested around setting custom brand with new image and description - no change, as before. Deferring manual testing of the error on the test server as this is mostly fixing a failing test.

**Comment by jayaddison-collabora** [comment]
Thank you, @hahmadia @jgilliam17 @Willyfrog!

**Pull request merged**